ir/attrs.h

```diff
@@ -46,14 +48,16 @@ class FnAttrs final {
                    ArgMemOnly = 1 << 2, NNaN = 1 << 3, NoReturn = 1 << 4,
                    Dereferenceable = 1 << 5, NonNull = 1 << 6,
                    NoFree = 1 << 7, NoUndef = 1 << 8, Align = 1 << 9,
-                   NoThrow = 1 << 10, NoAlias = 1 << 11, WillReturn = 1 << 12 };
+                   NoThrow = 1 << 10, NoAlias = 1 << 11, WillReturn = 1 << 12,
+                   DereferenceableOrNull = 1 << 13 };
 
   FnAttrs(unsigned bits = None) : bits(bits) {}
 
   bool has(Attribute a) const { return (bits & a) != 0; }
   void set(Attribute a) { bits |= (unsigned)a; }
 
   uint64_t derefBytes; // Dereferenceable
+  uint64_t derefOrNullBytes; // DereferenceableOrNull
```

Comment:
This has to be a separate field to precisely encode the semantics of dereferenceable_or_null.
ex) `define i8* dereferenceable(1) dereferenceable_or_null(2) @f()`